edb/edgeql/compiler/casts.py

```diff
@@ -961,6 +961,17 @@ def _find_object_by_id(
                 right=uuid_anchor,
             ),
         )
-        ql = qlast.FunctionCall(func='assert_exists', args=[select_id])
+        ql = qlast.FunctionCall(
+            func='assert_exists',
+            args=[select_id],
+            kwargs={
+                'message': qlast.StringConstant(
```

Comment:
Can we make the error message here actually include the id?

Reply:
Yeah, that's a good question. Let's try benchmarking it.

Reply:
Otherwise I would go with something like
```
could not find <typename> with specified id
```
I think that including the id would be pretty nice, though. So if it is slow, it might be worth looking into way to make it not too bad?

Reply:
I've tried using this:
```
> time edb cli query "select (for i in _gen_series(1, 10000000000) union (select <Person><uuid>'b90b4f92-4331-11ed-adee-17dbc27f8b02' filter .first_name = 'non-existent'));"
edb cli query   0.52s user 0.06s system 99% cpu 0.589 total
```
and it from 0.49s - 0.55s, no matter if I have ids in error message or not.

So I'm keeping it in.